### DISCUSSION TITLE
Are there more detailed docs about MLIR in Mojo?

### DISCUSSION THREAD

**Discussion by czheo** [modular/modular#151] [Q&A] [answered]
Noticed MLIR code like `!pop...` / `!kgen...`, such as `!pop.simd` or `!kgen.dtype`. I couldn't find more information from Mojo/MLIR's doc. What are `pop` and `kgen`?
Also the syntax of MLIR in Mojo is quite obscure. Are there/will there be more detailed docs?

**Comment by sa-** [comment]
I am also interested here

**Comment by lattner** [comment, marked as answer]
The mojo compiler has a number of internal dialects, including pop and kgen, but they aren't documented yet.  They are very much internal implementation details of the compiler and change all the time.  I'd recommend sticking with the llvm and other dialects that are more stable.
